service/jwks_fetching_roundtripper.go

```diff
@@ -0,0 +1,25 @@
+package service
+
+import "net/http"
+
+// InternalErrorHandler is a type alias for a function that receives an error.
+// The caller can be notified about the internal errors of the package.
+type InternalErrorHandler func(error)
+
+// JWKSFetchingRoundTripper ...
+type JWKSFetchingRoundTripper struct {
+	ErrorHandler InternalErrorHandler
+}
+
+// RoundTrip ...
+func (roundTripper *JWKSFetchingRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	res, err := http.DefaultTransport.RoundTrip(req)
+
+	if err != nil {
+		if roundTripper.ErrorHandler != nil {
+			roundTripper.ErrorHandler(err)
+		}
+	}
+
+	return res, err
```

Comment:
sorry for the challenging question, but if we return `err` here, what's the point of the `ErrorHandler`? What benefits do we get from this? 🤔

Reply:
This task is about to create a way that consumers can be notified about `internal` (as we call in the scope of this task) errors, making it possible for them to distinguish between validation errors (like an expired or invalid token) and `internal` errors (like errors during fetching the `JWKS`). This round-tripper is used by the client that is used to fetch the `JWKS`, which is considered as an `internal` task since this is happening within the `auth0` library as an effect of calling `validator.ValidateRequest`.
By the time this error will be visible at the call site, it will be indistinguishable for the caller whether it was an error in the validation process or it was an `internal` error, in this exact case, like a wrong endpoint, or timeout. So the error is returned by the round-tripper in every case, but in order to separate these types I mentioned above, if the error was emitted by the client that is used to fetch the `JWKS`, it will also be handled by the `internalErrorHandler` which will only be called upon an `internal` error is occurred, so the consumers can be sure that if an error is coming through the `internalErrorHandler`, it is really an `internal` error, not an error of the validation process for instance.
I hope it makes sense, but feel free to elaborate if not.

Reply:
IMHO the clients do not want multiple ways of error handling every client has one goal: getting the error and log it. If they get the errors in any way that's fine for the first go. Later on we could use different error types for internal errors as well, but I doubt if any clients would ever interested in this package internal error behaviours. The goal is they capable to get the errors and log it. For me it seems it's already possible, but let's have a chat about this.

Reply:
Discussed offline. This separate error handler is not necessary anymore. `internal` errors will be marked using the round-tripper, so the clients can handle them as they wish in the error writer.